docs/source/basics/customizing_interface.md

```diff
@@ -148,3 +109,292 @@ Each key in `metadata.parameters` must be the name of one of the parameters of t
 * `batch_table_column`: Whether this parameter should have a column to itself in the batch table at the top of the parameters page,
 * `appearance_type`: Either `LatchAppearanceType.line` or `LatchAppearanceType.paragraph`, which style to render text inputs as.
 * `rules`: A list of `LatchRule`s which consist of a regular expression and a message. If provided, an input must match all given regexes in order to appear valid in the front end - if it fails to match one of the regexes, the corresponding message is displayed.
+
+See below for a parameter display that uses all options mentioned: 
+
+```python
+from latch.types import LatchMetadata, LatchAuthor, LatchRule, LatchAppearanceType
+
+metadata = LatchMetadata(
+    parameters={
+        "read1": LatchParameter(
+            display_name="Read 1",
+            description="Paired-end read 1 file to be assembled.",
+            hidden=True,
+            section_title="Sample Reads",
+            placeholder="Select a file",
+            comment="This is a comment",
+            output=False,
+            appearance_type=LatchAppearanceType.paragraph,
+            rules=[
+                LatchRule(
+                    regex="(.fasta|.fa|.faa|.fas)$",
+                    message="Only .fasta, .fa, .fas, or .faa extensions are valid"
+                )
+            ],
+            batch_table_column=True,  # Show this parameter in batched mode.
+        ),
+)
+
+@workflow(metadata)
+def wf(read1: LatchFile):
+    ...
+```
+
+### How Python types of paramters translate to the UI
+
+Latch parses the Python type of your workflow parameters to generate the appropriate interface. 
+
+Below is a list of examples of Python types and how they translate to the UI:
+
+1. LatchFile
+
+```python
+from latch.types import LatchFile
+from typing import Optional
+
+...
+@workflow
+def bactopia_wf(
+    ...
+    fastq_one: Optional[LatchFile] = None,
+    ...
+)
+```
```

Comment:
I would separate the types and things like optional and default values — and explain them separately. But could do this later